src/app/component/result/result.component.ts

```diff
@@ -9,46 +8,37 @@ import { Params, Router } from '@angular/router';
   styleUrls: ['./result.component.scss']
 })
 export class ResultComponent implements OnInit, OnDestroy {
-  quizData = { ...quizData };
   userName: any;
-  userData: any;
-  submittedData: any;
+  allResultData: any;
 
   constructor(
     public questionService: QuestionService,
     public router:Router
     ) {}
 
   ngOnInit(): void {
-    this.userName = this.questionService?.userName;
     this.resultData();
     if (!this.userName) {
       this.getData();
     }
   }
 
   getData() {
-    let data: any = localStorage.getItem('registeruser');
-    this.userData = JSON.parse(data);
-    this.userName = this.userData.find((data: any) => {
+    let data: any = localStorage.getItem('registerUser');
+    this.userName = JSON.parse(data).find((data: any) => {
       return data.id == this.questionService.getUser();
-    })?.fullname;
+    })?.fullName;
   }
 
   resultData() {
     let data: any = localStorage.getItem('result');
-    this.submittedData = JSON.parse(data);
-    this.submittedData = this.submittedData.filter((data: any) => {
+    this.allResultData = JSON.parse(data).filter((data: any) => {
       return data?.user == this.questionService.getUser();
     });
-    this.submittedData = this.submittedData.map((data: any) => {
-      const message = this.quizData?.quiz?.find((quizData: any) => {
-        return quizData.quizId == data.type;
-      });
-      data.image = message?.image;
-      return data;
-    });
-    this.submittedData = this.submittedData?.reverse();
+    this.allResultData = this.allResultData.splice(
+      this.allResultData.length - 1,
```

Comment:
why doing thing ? like minus with length ?

Reply:
instead can we peek like with date ? as what is latest data he gave the example do one thing dont make loop instead make object and access property instead

Reply:
yes i changed it and managed it with timestramp